File: src/gestorAplicacion/economia/Salida.java
package gestorAplicacion.economia;

import java.time.LocalDate;

import gestorAplicacion.usuario.Cuenta;

public class Salida extends Movimiento{

	/**
	 * 
	 */
	private static final long serialVersionUID = -3919824199311137700L;
	
	private Cuenta cuentaOrigen;
	private Cuenta cuentaDestino;
	private Divisa divisaOrigen;
	private Divisa divisaDestino;
	private Banco bancoDestino;
	
	public Salida(double valorDestino, double valorOrigen, LocalDate fechaCreacion, Cuenta cuentaOrigen, Cuenta cuantaDestino, Divisa divisaOrigen, Divisa divisaDestino) {
		super(valorDestino,valorOrigen, fechaCreacion, true, null);
		setCuentaOrigen(cuentaOrigen);
		setCuentaDestino(cuantaDestino);
		setDivisaOrigen(divisaOrigen);
		setDivisaDestino(divisaDestino);
	}
	
	public Salida(double valorDestino, LocalDate fechaCreacion, Cuenta cuentaOrigen, Banco bancoDestino, Divisa divisaOrigen) {
		super(valorDestino,valorDestino, fechaCreacion, false, bancoDestino);
		setBancoDestino(bancoDestino);
		setCuentaOrigen(cuentaOrigen);
		setDivisaOrigen(divisaOrigen);
	}
	
	public Cuenta getCuentaOrigen() {
		return cuentaOrigen;
	}
	public void setCuentaOrigen(Cuenta cuentaOrigen) {
		this.cuentaOrigen = cuentaOrigen;
	}
	public Cuenta getCuentaDestino() {
		return cuentaDestino;
	}
	public void setCuentaDestino(Cuenta cuentaDestino) {
		this.cuentaDestino = cuentaDestino;
	}
	public Divisa getDivisaOrigen() {
		return divisaOrigen;
	}
	public void setDivisaOrigen(Divisa divisaOrigen) {
		this.divisaOrigen = divisaOrigen;
	}
	public Divisa getDivisaDestino() {
		return divisaDestino;
	}
	public void setDivisaDestino(Divisa divisaDestino) {
		this.divisaDestino = divisaDestino;
	}
	public Banco getBancoDestino() {
		return bancoDestino;
	}
	public void setBancoDestino(Banco bancoDestino) {
		this.bancoDestino = bancoDestino;
	}
	
	
	
}

File: Python/gestorAplicacion/economia/Salida.py
from gestorAplicacion.economia import Movimiento

class Salida(Movimiento):

    def __init__(self, valorDestino, valorOrigen, fechaCreacion, cuentaOrigen, cuantaDestino, divisaOrigen, divisaDestino):
        super().__init__(valorDestino,valorOrigen, fechaCreacion, True, None)
        self._cuentaOrigen = cuentaOrigen
        self._cuentaDestino = cuantaDestino
        self._divisaOrigen = divisaOrigen
        self._divisaDestino = divisaDestino

    def getCuentaOrigen(self):
        return self._cuentaOrigen

    def setCuentaOrigen(self, cuentaOrigen):
        self._cuentaOrigen = cuentaOrigen

    def getCuentaDestino(self):
        return self._cuentaDestino

    def setCuentaDestino(self, cuentaDestino):
        self._cuentaDestino = cuentaDestino

    def getDivisaOrigen(self):
        return self._divisaOrigen

    def setDivisaOrigen(self, divisaOrigen):
        self._divisaOrigen = divisaOrigen

    def getDivisaDestino(self):
        return self._divisaDestino

    def setDivisaDestino(self, divisaDestino):
        self._divisaDestino = divisaDestino

    def getBancoDestino(self):
        return self._bancoDestino
        
    def setBancoDestino(self, bancoDestino):
        self._bancoDestino = bancoDestino


